File: Examples/test-suite/ocaml/funcptr_cpp_runme.ml
open Swig
open Funcptr_cpp

let _ =
  let fp = _ADD_BY_VALUE '() in
  assert (_call1 '(fp, 10, 11) as int = 21);
  let fp = _ADD_BY_POINTER '() in
  assert (_call2 '(fp, 12, 13) as int = 25);
  let fp = _ADD_BY_REFERENCE '() in
  assert (_call3 '(fp, 14, 15) as int = 29);
  let fp = _ADD_BY_VALUE_C '() in
  assert (_call1 '(fp, 2, 3) as int = 5);
;;

File: Examples/test-suite/lua/funcptr_cpp_runme.lua
require("import")
require("funcptr_cpp")
fc=funcptr_cpp
catch_undef_globs() -- catch "undefined" global variables

assert(fc.addByValue(5,10) == 15)
-- These two won't work. Lua will successfully store the answer as userdata, but there is
-- no way of accessing the insides of userdata.
-- assert(fc.addByPointer(7, 9) == 16)
-- assert(fc.addByReference(8, 9) == 17)

assert(fc.call1(fc.ADD_BY_VALUE, 5, 10) == 15)
assert(fc.call2(fc.ADD_BY_POINTER, 7, 9) == 16)
assert(fc.call3(fc.ADD_BY_REFERENCE, 8, 9) == 17)
assert(fc.call1(fc.ADD_BY_VALUE_C, 2, 3) == 5)
